### PR TITLE
Prestandafix

### PR DESCRIPTION
**Pull request by agnepagne**
Making ABM table functions a bit faster by delaying `collect()` until group summaries are done, where possible. The slowest one, `abm_table1()`, improves the most and is ~3 times faster for KTH, the effect is smaller on smaller units since the speedup is mainly due to not fetching many rows from database. The other tables are 1.5-2 times faster.

This should make updating of `abm_public_kth` data considerably faster, hopefully avoiding database timeout, and on-the-fly dashboard generation for individuals slightly faster.

Other changes:
- No default for database connection, so always call with a connection from outside of function
- No longer closing database connection inside function
- No dependency on the `abm_data()` function anywhere in the package, so we can remove it if we want to
- No run_unit in abm.Rmd, use params$unit_code to select unit to run.

The tests in test_abm_full.R still pass, so nothing should be broken ...

### CODE REVIEW

File: R/abm.R [hunk 1]
```diff
@@ -143,7 +150,7 @@ abm_table2 <- function(con = con_bib(), unit_code, analysis_start = abm_config()
 #' @param last the largest integer in the range to use
 #' @param width the desired width of intervals
 #' @return data frame with label for each interval and one row for each year
-
+#' @export
 sliding_intervals <- function(first, last, width){
```
Comment (mskyttner) [inline R/abm.R]: Not important, I guess, but this fcn could return a tibble since we're doing that elsewhere?

File: R/abm.R [hunk 2]
```diff
@@ -269,18 +280,21 @@ abm_table4 <- function(con = con_bib(), unit_code, analysis_start = abm_config()
...
+             Publication_Year <= analysis_stop &
+             Publication_Type_WoS %in% c("Article", "Review") &
+             !is.na(int))
 
   # Duplicate rows so that publications are connected to all intervals they should belong to according to publication year
   orgdata3year <-
-    orgdata %>% 
+    orgdata %>%
+    collect() %>% 
     inner_join(sliding_intervals(analysis_start, analysis_stop, 3),
```
Comment (mskyttner) [inline R/abm.R]: This is not important really but I saw this package "funneljoin" which is now on CRAN which seems to have some nice join functions for time series data kept in data frames: https://hookedondata.org/introducing-the-funneljoin-package/

Comment (mskyttner) [reply R/abm.R]: Also, again not relevant here, but more stuff for tidy work with timeseries: https://tidyverts.org/

Comment (agnepagne) [reply R/abm.R]: Thanks for the links @mskyttner, both seem interesting! Maybe we can have a closer look later, when there is more time for refactoring.

File: R/abm.R (hunk 1 above)
Comment (agnepagne) [reply R/abm.R]: Seems reasonable. :) Do we have a general "to do when we have time to refactor" issue somewhere?

**Pull request merged**